src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,143 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_checkOnUserRegistration_isOk() {
+        User actual = new User();
+        actual.setLogin("Snowtrack");
+        actual.setPassword("1234567");
+        actual.setAge(44);
+        User actual1 = new User();
+        actual1.setLogin("Genadiy");
+        actual1.setPassword("1234567");
+        actual1.setAge(18);
+        User actual2 = new User();
+        actual2.setLogin("Nicole");
+        actual2.setPassword("1234567");
+        actual2.setAge(22);
+        User actual3 = new User();
+        actual3.setLogin("Snowhunter");
+        actual3.setPassword("123456");
+        actual3.setAge(41);
+        User expected = registrationService.register(actual);
+        assertEquals(expected, actual);
+        expected = registrationService.register(actual1);
+        assertEquals(expected, actual1);
+        expected = registrationService.register(actual2);
+        assertEquals(expected, actual2);
+        expected = registrationService.register(actual3);
+        assertEquals(expected, actual3);
+    }
```

Comment:
This test method is too long and checks too many cases. It's better to split it into several smaller test methods, each checking one specific case. This way, when a test fails, you'll know exactly what case is not working, just by looking at the test method's name.